### PR TITLE
NETBEANS-3547 Support vendor specific JVM options in Payara Server

### PR DESCRIPTION
**Pull request by MeroRai**
The code changes only affect Payara related code. It is a feature to support vendor specific JVM options in Payara Server.

### CODE REVIEW

File: enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/JDKVersion.java [hunk 1]
```diff
@@ -54,19 +59,27 @@
 
     private static final Short DEFAULT_VALUE = 0;
 
-    private JDKVersion(String string) {
-        String[] split = string.split(VERSION_SPLITTER);
+    private JDKVersion(String version) {
+        if (version.contains("-")) {
+            String[] versionSplit = version.split("-");
+            vendor = versionSplit.length > 0 ? Optional.of(versionSplit[0]) : Optional.empty();
+            version = versionSplit.length > 1 ? versionSplit[1] : "";
```
Comment (junichi11) [inline enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/JDKVersion.java]: Please add `// NOI18N`s. Strings of other files as well.
e.g.
```java
        if (version.contains("-")) { // NOI18N
            String[] versionSplit = version.split("-"); // NOI18N
            vendor = versionSplit.length > 0 ? Optional.of(versionSplit[0]) : Optional.empty();
            version = versionSplit.length > 1 ? versionSplit[1] : ""; // NOI18N
```

File: enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/JDKVersion.java [hunk 2]
```diff
@@ -235,19 +257,23 @@ public static JDKVersion getDefaultPlatformVersion() {
...
+        if (vendor.isPresent()) {             
+            correctJDK = jdkVersion.vendor.get().contains(vendor.get());
         }
+        if (correctJDK && minVersion.isPresent()) {
+            correctJDK = jdkVersion.ge(minVersion.get());
+        }       
         if (correctJDK && maxVersion.isPresent()) {
             correctJDK = jdkVersion.le(maxVersion.get());
-        }
+        }
```
Comment (junichi11) [inline enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/JDKVersion.java]: Please remove trailing spaces.

File: enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/JDKVersion.java [hunk 3]
```diff
@@ -259,6 +285,7 @@ private static void initialize() {
         short minor = 0;
         short subminor = 0;
         short update = 0;
+        String vendor = System.getProperty("java.vendor");
```
Comment (junichi11) [inline enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/JDKVersion.java]: Please add `// NOI18N`.

File: enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/server/parser/JvmConfigReader.java [hunk 4]
```diff
@@ -217,18 +218,27 @@ public boolean isMonitoringEnabled() {
         public JvmOption(String option) {
             Matcher matcher = PATTERN.matcher(option);
             if (matcher.matches()) {
-                this.minVersion = Optional.ofNullable(JDKVersion.toValue(matcher.group(1)));
+                if (matcher.group(1).contains("-")) {
+                    String[] parts = matcher.group(1).split("-");
```
Comment (junichi11) [inline enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/server/parser/JvmConfigReader.java]: Please add `// NOI18N`s.

**Comment by junichi11** [comment]
@MeroRai Maybe, should submit an issue to the JIRA if possible before you create a PR: https://issues.apache.org/jira/projects/NETBEANS/issues

Then, please also see this: https://cwiki.apache.org/confluence/display/NETBEANS/Submitting+Pull+Request+on+Apache+NetBeans

Thank you for your contribution :)

**Comment by lkishalmi** [comment]
Well, I think our utilities module woule welcome a such a JDK version parsing class.
Please consider implementing the Comparable interface. The IDE_JDK_VERSION shall be final or private (as you have an access method for it).

**Comment by jGauravGupta** [comment]
Hi @lkishalmi,

JDK version parsing class parses the proprietary version formate defined in domain.xml of Payara Server, hence it is not suitable for the utilities module.

**Comment by junichi11** [comment]
@MeroRai Thank you for fixing them.

@jGauravGupta If it's OK with you, please approve it. Let's squash and merge :)

**Comment by matthiasblaesing** [comment]
@jGauravGupta you can merge this yourself. However be careful with the "Squash and merge" option in the github GUI. If the account of the author is not configured correctly, it will create invalid author information. In this case @MeroRai does not have a public email address configured for this account, so this will create a squash commit without a valid email.

In this case please squash outside github (IMHO the github UI is useless as is) and push the squashed commit directly. When you squash manually the author information will be retained, you will be marked as committer.

**Comment by jGauravGupta** [comment]
Hi,
Merge Commits failed the `Test platform modules, Batch 2` job but the squashed commits passed, Anyone knows what went wrong?

**Comment by junichi11** [comment]
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```

Will try restarting.

**Comment by junichi11** [comment]
Passed.

**Pull request merged**